src/mbgl/text/shaping.cpp

```diff
@@ -92,6 +92,45 @@ PositionedIcon PositionedIcon::shapeIcon(const ImagePosition& image,
     return PositionedIcon{image, top, bottom, left, right, collisionPadding};
 }
 
+PositionedIcon PositionedIcon::applyTextFit() const {
+    if (!_image.textFitWidth && !_image.textFitHeight) {
+        return *this;
+    }
+    auto width = _right - _left;
+    auto height = _bottom - _top;
+
+    auto contentWidth = _image.content->right - _image.content->left;
```

Comment:
Please have a look over all definitions in the PR and make sure that all immutable values are marked `const`.

Reply:
Done.